src/Controller/EloquaAppCloudEndpointController.php

```diff
@@ -0,0 +1,428 @@
+      ->load($eloqua_app_cloud_service);
+    // Return the list of plugin references.
+    return $entity->field_eloqua_app_cloud_responder->getIterator();
+  }
+
+  /**
+   * @param $pluginReferences
+   *
+   * @return array
+   */
+  private
+  function getFieldList($pluginReferences) {
+    // Iterate over the ServiceEntity plugins and build a merged field list.
+    $fieldLists = [];
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+      // We can only allow ONE type of API at a time.
+      if (!empty($api) && $api !== $plugin->api()) {
+        // Throw an exception!
+      }
+      $api = $plugin->api();
+      // Merge the required fieldlists of all the listed plugins.
+      $fieldLists = array_merge($fieldLists, $plugin->fieldList());
+    }
+    return $fieldLists;
+  }
+
+  /**
+   * Returns HTML to be displayed in a popup configure window on the Eloqua
+   * canvas. If required it should also return an bulk API request to update
+   * the field list for this service.
+   *
+   * @TODO: implement a way to request an updated field list :)
+   *   (https://github.com/tableau-mkt/eloqua_app_cloud/issues/3)
+   *
+   * @param $eloqua_app_cloud_service
+   *
+   * @return mixed
+   */
+  public function update($eloqua_app_cloud_service) {
+    $query = $this->request->query->all();
+    // Get the instanceID from the query parameter.
+    $instanceId = $query["instance"];
+    if (empty($instanceId)) {
+      $this->logger->error('No instanceID found');
+      // It apparently does not matter what we return here, since Eloqua does not have any way of handing external errors?
+      return new HtmlResponse('No instanceID found.', 500);
+    }
+    $pluginReferences = $this->getEntityPlugins($eloqua_app_cloud_service);
+
+    $response = [
+      '#theme' => 'eloqua_app_cloud_update',
+    ];
+    // Punting on the case where there are multiple plugins for a service.
+    // This code will always return all the instantiate responses for the
+    // last plugin in the list, but does not try to help if they break each other.
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+      $update = $plugin->update($instanceId, $query);
+      $response['#content'][$plugin->getPluginId()] = $update;
+    }
+    return $response;
+  }
+
+  /**
+   * Delete any existing queue entries for this service entity.
+   *
+   * @TODO: Call the plugins deete method to let it do specific cleanup instead
```

Comment:
nit: Typo in your todo. But is there any harm in just calling it here now?  Seems like you're already instantiating a plugin instance below.